lib/vdsm/storage/exception.py

```diff
@@ -115,6 +115,22 @@ def __init__(self, volume, *args):
             self.value += list(args)
 
 
+class VolumeLVMCommandError(StorageException):
+    code = 4001
+    msg = "Volume exception with LVM error"
+
+    def __init__(self, error=None):
+        if error is not None and not isinstance(error, LVMCommandError):
+            raise TypeError(
+                f"Expecting instance of LVMCommandError, got {type(error)}")
+
+        self.error = error
+
+    @property
+    def value(self):
+        return ", ".join(f"{k}={v}" for k, v in self.__dict__.items() if v)
```

Comment:
Nice that we have one place to format instead of repeating this in 3 classes.

It would be nice if we use this for all storage errors in the future.

Reply:
Sounds good. I'll create an issue.

I have a few notes taken that need to be written into issues...